RFS/src/main/java/com/rfs/worker/GlobalState.java

```diff
@@ -0,0 +1,52 @@
+package com.rfs.worker;
+
+import java.util.concurrent.atomic.AtomicReference;
+
+/*
+ * This class is a Singleton that contains global process state.
+ */
+public class GlobalState {
+    private static final AtomicReference<GlobalState> instance = new AtomicReference<>();
+
+    enum Phase {
+        UNSET,
+        SNAPSHOT_IN_PROGRESS,
+        SNAPSHOT_COMPLETED,
+        SNAPSHOT_FAILED,
+        METADATA_IN_PROGRESS,
+        METADATA_COMPLETED,
+        METADATA_FAILED
+    }    
+
+    private AtomicReference<Phase> phase = new AtomicReference<>(Phase.UNSET);
+    private AtomicReference<OpenSearchWorkItem> workItem = new AtomicReference<>(null);
+
+    private GlobalState() {}
+
+    public static GlobalState getInstance() {
+        instance.updateAndGet(existingInstance -> {
+            if (existingInstance == null) {
+                return new GlobalState();
+            } else {
+                return existingInstance;
+            }
+        });
+        return instance.get();
+    }
+
+    public void updatePhase(Phase newValue) {
```

Comment:
Can/Should this take in SnapshotStatus or MetadataStatus and apply the conversion to phase here?